fusor-ember-cli/app/mixins/polling-promise-mixin.js

```diff
@@ -0,0 +1,46 @@
+import Ember from 'ember';
+
```

Comment:
The progress-bar mixin is executing an action every X seconds.  I was having issues (especially when first loading a page) that multiple requests would stack up and not allow the previous requests to finish.  Instead of using actions, this one calls a function that returns a promise and doesn't start the timer until the action is complete.  That way if the request takes 10 seconds on a 5 second interval, multiple requests won't stack up.  I was also anticipating needing to poll multiple things with different names, so I added that, but ended up not needing to do multiples but kept the naming customization in.